packages/block/src/blockchainMachine.ts

```diff
@@ -58,65 +63,47 @@ export class BlockchainMachine<TBlock extends IBlockStub> {
+     * If the state of the parent is not present, it is computed by using `getInitialState` on the reducer. This should only happen at
+     * startup. therefore, an error is logged if this happens after some previous anchor states were previously stored in the blockItemStore.
+     * If the state of the parent is present, then the state for the current block is computed by using the `reduce` function of the reducer.
      * Once a new state is set changes are detected between this and the parent state, and actions are computed and executed.
      * This function writes to the blockItemStore and MUST be included in a batch to persist beyond memory.
      * @param block
      */
     public async setStateAndDetectChanges(block: TBlock) {
         if (!this.actionStore.started) logger.error("The actionStore should be started before the BlockchainMachine is used.");
         if (!this.blockItemStore.started) logger.error("The BlockItemStore should be started before the BlockchainMachine is used.");
-        
+
         try {
             await this.lock.acquire();
 
+            const isFirstAnchorState = !this.blockItemStore.isAnchorStateInitialised;
+
             // Every time a new block is received we calculate the anchor state for that block and store it
             for (const component of this.components) {
                 // get the parent
                 const parentState = this.blockItemStore.anchorState.get<AnchorState>(component.name, block.parentHash);
-                if (!parentState) throw new ConfigurationError(`Parent state not already set for component: ${component.name} block: ${block.number - 1}:${block.parentHash}`); //prettier-ignore
 
-                // reduce the new state
-                const newAnchorState = await component.reducer.reduce(parentState, block);
+                let newAnchorState: AnchorState;
+                if (parentState != undefined) {
+                    newAnchorState = await component.reducer.reduce(parentState, block);
+                } else {
+                    if (isFirstAnchorState) this.logger.error({ componentName: component.name }, "Did not find anchor state for component, but it was expected.");
+
+                    newAnchorState = await component.reducer.getInitialState(block);
+                }
+
                 this.blockItemStore.anchorState.set(component.name, block.number, block.hash, newAnchorState);
 
-                // having computed a new state we can detect changes and run actions
-                // for the difference between parent and now
-                const newActions = component.detectChanges(parentState, newAnchorState);
-                if (newActions.length > 0) {
-                    const actionAndIds = await this.actionStore.storeItems(component.name, newActions);
-                    this.runActionsForComponent(component, actionAndIds);
+                if (parentState != undefined) {
```

Comment:
Rewritten in 31aeed1 to ofix this and other issues.

There is a bit of code dupllication because I added some conservative behavior if the parent is not found when expected (log an error but still try to compute a meaningful state, rather than giving up).